src/sas/sascalc/dataloader/readers/sesans_reader.py

```diff
@@ -17,158 +17,161 @@
-                zaccept_unit = zaccept_unit_split[1].replace("]","")
-                if zaccept_unit.strip() == r'\AA^-1' or zaccept_unit.strip() == r'\A^-1':
-                    zaccept_unit = "1/A"
-                output.sample.zacceptance=(float(paramvals[7]),zaccept_unit)
-                output.vars = varheader
+                    line = input_f.readline()
+                    params = {}
+                    while line.strip() != "":
+                        terms = line.strip().split("\t")
+                        params[terms[0].strip()] = " ".join(terms[1:]).strip()
+                        line = input_f.readline()
+                    headers_temp = input_f.readline().strip().split("\t")
+                    headers = {}
+                    for h in headers_temp:
+                        temp = h.strip().split()
+                        headers[h[:-1].strip()] = temp[-1][1:-1]
+                    data = np.loadtxt(input_f)
+                    x = data[:, 0]
+                    dx = data[:, 3]
+                    lam = data[:, 4]
+                    dlam = data[:, 5]
+                    y = data[:, 1]
+                    dy = data[:, 2]
+
+                    lam_unit = self._header_fetch(headers, "wavelength")
+                    if lam_unit == "AA":
+                        lam_unit = "A"
+
+                    x, x_unit = self._unit_conversion(
+                        x, lam_unit,
+                        self._fetch_unit(headers, "spin echo length"))
+                    dx, dx_unit = self._unit_conversion(
+                        dx, lam_unit,
+                        self._fetch_unit(headers, "error SEL"))
+                    dlam, dlam_unit = self._unit_conversion(
+                        dlam, lam_unit,
+                        self._fetch_unit(headers, "error wavelength"))
+                    y_unit = r'\AA^{-2} cm^{-1}'
+
+                    output = Data1D(x=x, y=y, lam=lam, dy=dy, dx=dx, dlam=dlam,
+                                    isSesans=True)
+                    self.filename = output.filename = basename
+                    output.xaxis(r"\rm{z}", x_unit)
+                    # Adjust label to ln P/(lam^2 t), remove lam column refs
+                    output.yaxis(r"\rm{ln(P)/(t \lambda^2)}", y_unit)
+                    # Store loading process information
+                    output.meta_data['loader'] = self.type_name
+                    output.sample.name = params["Sample"]
+                    output.sample.ID = params["DataFileTitle"]
+
+                    output.sample.zacceptance = (
+                        float(self._header_fetch(params, "Q_zmax")),
+                        self._fetch_unit(params, "Q_zmax"))
+
+                    output.sample.yacceptance = (
+                        float(self._header_fetch(params, "Q_ymax")),
+                        self._fetch_unit(params, "Q_ymax"))
 
                 if len(output.x) < 1:
```

Comment:
That empty file check really needed to be done right after the data was read.  I've moved it there in commit 56e2e3e